### PR TITLE
Add VIA code to ergoTravel, VendorId changed to allow Unique Keyboard Identification

### PR DESCRIPTION
**Pull request by harshitgoel96**
In my opinion every keyboard that runs QMK should have VIA support.  I have added VIA configuration changes to ErgoTravel Keyboard. Tagging @jpconstantineau original hardware maintainer.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [x] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by harshitgoel96** [comment]
Via Design lab is required to make it work.

File: keyboards/ergotravel/keymaps/via/keymap.c [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+  //     if (record->event.pressed) {
+  //       layer_on(_ADJUST);
+  //     } else {
+  //       layer_off(_ADJUST);
+  //     }
+  //     return false;
+  //     break;
+  // }
+  // return true;
+// }
```
Comment (fauxpark) [inline keyboards/ergotravel/keymaps/via/keymap.c:113]:
```suggestion
```

File: keyboards/ergotravel/rules.mk [hunk 2]
```diff
@@ -15,21 +15,23 @@ BOOTLOADER = caterina
...
 AUDIO_ENABLE = no           # Audio output on port C6
 UNICODE_ENABLE = no         # Unicode
 BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
-RGBLIGHT_ENABLE = yes       # Enable WS2812 RGB underlight. 
+RGBLIGHT_ENABLE = no       # Enable WS2812 RGB underlight. 
 # Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
 SLEEP_LED_ENABLE = no    # Breathing sleep LED during USB suspend
 
 SPLIT_KEYBOARD = yes
+VIA_ENABLE = yes
```
Comment (fauxpark) [inline keyboards/ergotravel/rules.mk:34]: This does not go at keyboard level. Please make these changes in the keymap rules.mk.

File: keyboards/ergotravel/via.json [hunk 3]
```diff
@@ -0,0 +1,16 @@
+{
```
Comment (fauxpark) [inline keyboards/ergotravel/via.json:1]: This file also does not belong here.

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/ergotravel/via.json (hunk 3 above)
Comment (Wilba6582) [reply keyboards/ergotravel/via.json:1]: Submit a PR to https://github.com/the-via/keyboards

Comment (harshitgoel96) [reply keyboards/ergotravel/via.json:1]: working on the changes.

**Comment by harshitgoel96** [comment]
All change request from above implemented.

via pull request link
https://github.com/the-via/keyboards/pull/347

File: keyboards/ergotravel/keymaps/via/rules.mk [hunk 4]
```diff
@@ -0,0 +1,4 @@
+#VIA Specific rules.
+BOOTMAGIC_ENABLE = lite
```
Comment (fauxpark) [inline keyboards/ergotravel/keymaps/via/rules.mk:2]:
```suggestion
```
`VIA_ENABLE = yes` implies `BOOTMAGIC_ENABLE = lite`

File: keyboards/ergotravel/rules.mk [hunk 5]
```diff
@@ -15,21 +15,22 @@ BOOTLOADER = caterina
...
 UNICODE_ENABLE = no         # Unicode
 BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
-RGBLIGHT_ENABLE = yes       # Enable WS2812 RGB underlight. 
+RGBLIGHT_ENABLE = no       # Enable WS2812 RGB underlight. 
 # Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
 SLEEP_LED_ENABLE = no    # Breathing sleep LED during USB suspend
 
 SPLIT_KEYBOARD = yes
 
+
```
Comment (fauxpark) [inline keyboards/ergotravel/rules.mk:35]:
```suggestion
```

File: keyboards/ergotravel/rules.mk [hunk 6]
```diff
@@ -15,21 +15,22 @@ BOOTLOADER = caterina
...
-BOOTMAGIC_ENABLE = no       # Virtual DIP switch configuration
+BOOTMAGIC_ENABLE = no        # Virtual DIP switch configuration
 MOUSEKEY_ENABLE = yes       # Mouse keys
 EXTRAKEY_ENABLE = yes       # Audio control and System control
 CONSOLE_ENABLE = no         # Console for debug
-COMMAND_ENABLE = yes        # Commands for debug and configuration
+COMMAND_ENABLE = no        # Commands for debug and configuration
 NKRO_ENABLE = no            # Nkey Rollover - if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
-BACKLIGHT_ENABLE = yes      # Enable keyboard backlight functionality
+BACKLIGHT_ENABLE = no      # Enable keyboard backlight functionality
```
Comment (fauxpark) [inline keyboards/ergotravel/rules.mk:24]: Why change these? Underglow and backlight are both supported by VIA.

File: keyboards/ergotravel/rules.mk [hunk 7]
```diff
@@ -15,21 +15,22 @@ BOOTLOADER = caterina
 #   change to "no" to disable the options, or define them in the Makefile in
 #   the appropriate keymap folder that will get included automatically
 #
-BOOTMAGIC_ENABLE = no       # Virtual DIP switch configuration
+BOOTMAGIC_ENABLE = no        # Virtual DIP switch configuration
```
Comment (fauxpark) [inline keyboards/ergotravel/rules.mk:18]:
```suggestion
BOOTMAGIC_ENABLE = no       # Virtual DIP switch configuration
```

File: keyboards/ergotravel/keymaps/via/rules.mk [hunk 8]
```diff
@@ -0,0 +1,4 @@
...
+VIA_ENABLE = yes
+LTO_ENABLE = yes
```
Comment (fauxpark) [inline keyboards/ergotravel/keymaps/via/rules.mk:4]:
```suggestion
LTO_ENABLE = yes
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/ergotravel/rules.mk (hunk 6 above)
Comment (harshitgoel96) [reply keyboards/ergotravel/rules.mk]: I do not use backlight or rgb, so made changes to save on space. have restored the default rules from qmk master.

File: keyboards/ergotravel/rules.mk [hunk 9]
```diff
@@ -15,21 +15,21 @@ BOOTLOADER = caterina
...
-BOOTMAGIC_ENABLE = no       # Virtual DIP switch configuration
-MOUSEKEY_ENABLE = yes       # Mouse keys
-EXTRAKEY_ENABLE = yes       # Audio control and System control
-CONSOLE_ENABLE = no         # Console for debug
-COMMAND_ENABLE = yes        # Commands for debug and configuration
+BOOTMAGIC_ENABLE = no		# Virtual DIP switch configuration
+MOUSEKEY_ENABLE = yes		# Mouse keys
+EXTRAKEY_ENABLE = yes		# Audio control and System control
+CONSOLE_ENABLE = no			# Console for debug
+COMMAND_ENABLE = yes		# Commands for debug and configuration
```
Comment (fauxpark) [inline keyboards/ergotravel/rules.mk:22]:
```suggestion
BOOTMAGIC_ENABLE = no       # Virtual DIP switch configuration
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = yes        # Commands for debug and configuration
```

File: keyboards/ergotravel/rules.mk [hunk 10]
```diff
@@ -15,21 +15,21 @@ BOOTLOADER = caterina
...
 NKRO_ENABLE = no            # Nkey Rollover - if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
-BACKLIGHT_ENABLE = yes      # Enable keyboard backlight functionality
+BACKLIGHT_ENABLE = yes		# Enable keyboard backlight functionality
```
Comment (fauxpark) [inline keyboards/ergotravel/rules.mk:24]:
```suggestion
BACKLIGHT_ENABLE = yes      # Enable keyboard backlight functionality
```

File: keyboards/ergotravel/rules.mk [hunk 11]
```diff
@@ -15,21 +15,21 @@ BOOTLOADER = caterina
...
 MIDI_ENABLE = no            # MIDI controls
 AUDIO_ENABLE = no           # Audio output on port C6
 UNICODE_ENABLE = no         # Unicode
 BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
-RGBLIGHT_ENABLE = yes       # Enable WS2812 RGB underlight. 
+RGBLIGHT_ENABLE = yes		# Enable WS2812 RGB underlight.
```
Comment (fauxpark) [inline keyboards/ergotravel/rules.mk:29]:
```suggestion
RGBLIGHT_ENABLE = yes       # Enable WS2812 RGB underlight. 
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/ergotravel/keymaps/via/keymap.c [hunk 12]
```diff
@@ -0,0 +1,55 @@
...
+     _______, _______,  _______, _______, _______, _______, _______,         _______,  _______, _______, _______, _______, _______, _______,
+     _______, _______, _______, _______,           _______, _______,         _______,  _______,          _______, _______, _______, _______
+  ),
+  [_FINL] = LAYOUT(
+     _______,  _______,  _______, _______, _______, _______, _______,         _______,  _______, _______, _______, _______, _______, _______,
+     _______, _______,  _______, _______, _______, _______, _______,         _______,  _______, _______, _______, _______, _______, _______,
+     _______, _______,  _______, _______, _______, _______, _______,         _______,  _______, _______, _______, _______, _______, _______,
+     _______, _______, _______, _______,           _______, _______,         _______,  _______,          _______, _______, _______, _______
+  )
+
```
Comment (fauxpark) [inline keyboards/ergotravel/keymaps/via/keymap.c:53]: One last thing, I think: this layer should be removed, or `DYNAMIC_KEYMAP_LAYER_COUNT` set to 5 in the via keymap config.h.

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by zvecr — CHANGES_REQUESTED** [review]
Also, the `VENDOR_ID/PRODUCT_ID`pair is non unique and shared with at least the `qwertyydox`

**Comment by harshitgoel96** [comment]
Is there anyway to check for or directly generate unique vid/pid ?

**Comment by fauxpark** [comment]
https://yanfali.github.io/qmk_usb_usage/

**Review by fauxpark — APPROVED** [review]

File: keyboards/ergotravel/keymaps/via/keymap.c [hunk 13]
```diff
@@ -0,0 +1,48 @@
+#include QMK_KEYBOARD_H
```
Comment (noroadsleft) [inline keyboards/ergotravel/keymaps/via/keymap.c:1]: Could you add a license header here, please?

Here's a sample:

https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/ergotravel/keymaps/via/keymap.c [hunk 14]
```diff
@@ -0,0 +1,48 @@
...
+
+
+#define _QWERTY 0
+#define _LOWER 1
+#define _RAISE 2
+#define _ADJUST 3
```
Comment (noroadsleft) [inline keyboards/ergotravel/keymaps/via/keymap.c:7]: Optional, but you could use an `enum` here to manage the layer indexes.

```suggestion
enum layer_names {
    _QWERTY,
    _LOWER,
    _RAISE,
    _ADJUST
};
```

File: keyboards/ergotravel/rev1/config.h [hunk 15]
```diff
@@ -18,8 +18,8 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
 #pragma once
 
 /* USB Device descriptor parameter */
-#define VENDOR_ID       0xCEEB
-#define PRODUCT_ID      0x1256
+#define VENDOR_ID       0x4A50
+#define PRODUCT_ID      0x4554
```
Comment (noroadsleft) [inline keyboards/ergotravel/rev1/config.h:22]:
```suggestion
#define VENDOR_ID       0x4A50  // "JP"
#define PRODUCT_ID      0x4554  // "ET"
```

**Review by noroadsleft — CHANGES_REQUESTED** [review]

**Comment by harshitgoel96** [comment]
implemented changes suggested by @noroadsleft

**Comment by noroadsleft** [comment]
Thanks!

**Comment by a8t** [comment]
hey @harshitgoel96 -- really appreciate the work you did here. Do you know how hard it would be to add RGB support? I'm willing to contribute some time, but I'm not usually a C developer.

**Comment by harshitgoel96** [comment]
> hey @harshitgoel96 -- really appreciate the work you did here. Do you know how hard it would be to add RGB support? I'm willing to contribute some time, but I'm not usually a C developer. Is it as simple as adding it to rules.mk?

Hi.
I am currently occupied with work. There is no coding required, it's just config.
I don't have RGB on my ergotravel hence I did not add it. I will try to look into it over weekend, but will need your support to test it out.

**Pull request merged**